src/views/check/results/results.html

```diff
@@ -82,28 +73,79 @@ <h1 class="govuk-heading-l">
       {{ govukTabs({ items: items })}}
 
     </div>
-  </div>
-
-  <div class="govuk-grid-row">
-    <div class="govuk-grid-column-two-thirds">
-
-      <form novalidate method="post">
-        <h3 class="govuk-heading-m">Before you continue</h3>
-
-        <p>Make sure your data is correct.</p>
-
-        <p>If you see anything wrong with your data, you can check it again.</p>
-
+  </div>  
+
+  {% if options.passedChecks.length > 0 %}
+    <div class="govuk-grid-row" id="passed-checks">
+      <div class="govuk-grid-column-two-thirds">
+        <h2 class="govuk-heading-m">Data that passed checks</h2>
+    
+        {{ govukTaskList({
+          idPrefix: "passed-checks",
+          items: options.passedChecks
+        }) }}
+      </div>
+    </div>
+  {% endif %}
+  
+  {% if options.tasksBlocking.length > 0 %}
+    <div class="govuk-grid-row" id="required-checks">
+      <div class="govuk-grid-column-two-thirds">
+        <h2 class="govuk-heading-m">Issues you must fix before submitting</h2>
+    
+        <p>You cannot submit your data to the platform until you fix these issues.</p>
+    
+        {{ govukTaskList({
+          idPrefix: "must-fix",
+          items: options.tasksBlocking
+        }) }}
+      </div>
+    </div>
+  {% endif %}
+  
+  {% if options.tasksNonBlocking.length > 0 %}
+    <div class="govuk-grid-row" id="optional-checks">
+      <div class="govuk-grid-column-two-thirds">
+        <h2 class="govuk-heading-m">Issues you can fix after submitting</h2>
+    
+        <p>You can submit your data with these issues but should fix them over time to improve the quality and usability of
+          the
+          data.</p>
+    
+        {{ govukTaskList({
+          idPrefix: "fix-later",
+          items: options.tasksNonBlocking
+        }) }}
+      </div>
+    </div>
+  {% endif %}
+  
+  {% if options.tasksBlocking.length > 0 %}
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+        <p>Fix the issues, then upload a new version of your data to check if it is ready to submit.</p>
+        
         {{ govukButton({
-          text: "Continue"
+          text: "Upload a new version",
+          href: "/check/upload-method"
         }) }}
-
-        <input type="hidden" name="dataLooksCorrect" value="yes">
-        <p><a href="{{ checkToolLink  }}">Check your updated data</a></p>
-
-      </form>
+      </div>
     </div>
-  </div>
+  {% else %}
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+        <div class="govuk-button-group">
+          <form method="post">
+            {{ govukButton({
+              text: "Continue"
+            }) }}
+          </form>
+
+          <a href="/check/upload-method">Upload a new version</a>
+        </div>
```

Comment:
The `<form>` tag needs to wrap around the button group div and the link needs a class of `govuk-link` for the styles to be applied correctly